docs/howtoguides/enable_landscape.rst

```diff
@@ -0,0 +1,47 @@
+How to enable Landscape
+***********************
+
+``pro`` offers a convenient way to register a machine with Landscape via the
+``pro enable landscape`` command.
+
+Enable interactively
+====================
+
+To register your machine by interactively providing your Landscape account
+details at the CLI, run:
+
+.. code:: bash
+
+   sudo pro enable landscape
+
+That command will install ``landscape-client`` and drop you into an interactive
+wizard to complete the Landscape registration for the machine.
+
+Enable non-interactively
+========================
+
+If you know the details of your Landscape setup and would like to register a
+machine without any interaction necessary, you can pass ``landscape-config``
+parameters to ``pro enable landscape`` after a ``--``. You should also use the
+``--assume-yes`` flag to ensure there are no prompts.
```

Comment:
One question I have (and is probably not really a problem for us but still might occur to a reader) is - if we can and should avoid the prompts by using --assume-yes, why are the prompts there by default? Surely it should have been that yes is assumed by default and that you can turn that off by doing --assume-no. Can you --assume-no? What is assume-yes actually doing? (I assume it's not just suppressing prompts, but is rather doing something else - I think it's better to be clear here to avoid raising questions). (also sorry, that's way more than one question)

Reply:
We only _should_ avoid prompts if you want a completely non-interactive enablement. The prompts are there in the `landscape-config` command that we don't own/control. 

Prompts can be useful - if you provide some but not all required parameters to landscape-config, it will prompt you for the other required fields. If you miss a required parameter with `--assume-yes` then it will just fail. The "interactive but with some parameters supplied" is a valid invocation of the command, just not something I think is worth recommending - so I didn't document it here.

The "problem" is that `landscape-config` currently prompts for non-required parameters, and that is why we _have_ to recommend `--assume-yes` here. That is arguably a UX problem with `landscape-config` but that is out of our control (and the landscape team is aware of it).

What `--assume-yes` is really doing is accepting defaults for those non-required parameters - that should probably be the description instead of "ensure there are no prompts".

Reply:
Understood, thanks for the context there. I think that given this situation, the edit you made is the best one that could be made. It neatly sidesteps the entire thorny issue - good job :)